core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,147 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for machine translation."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import utils
+
+(base_models,) = models.Registry.import_models(
+    [models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MachineTranslatedTextModel(base_models.BaseModel):
+    """Model for storing machine translations. Model instances have a key
+    generated from the source and target language codes, followed by a sha-1
+    hash of the origin text formated as follows:
+
+        [source_language_code]:[target_language_code]:[hashed_origin_text]
```

Comment:
Is the intention with this ID mapping to reuse translations if the text is the same? That might be fine for machine translations, but for manual translations we should also factor in the context since translations can change for different contexts even when the original English text is the same.

Reply:
Yes, this storage model exists to prevent hitting the Cloud Translate API more than once for the same translation. It is only relevant for machine translations which are displayed as a helper in the contributor dashboard translation opportunity modal. See the [Computer Aided Translation project](https://docs.google.com/document/d/1kJd-yLTzB9a2c3Nq7v9pzKfHwKHKGpkWfQ8B0YGf50U/edit?usp=sharing) for more details.

Reply:
Hmm, I think it might be worth indicating in the model's docstring that models are expected to be mapped this way (e.g. that it's okay we're using the same translation irrespective of context because of relevant reasons, maybe: these translations are only meant to be used as a reference, and we want to avoid hitting the translation API multiple times since it will presumably provide consistent translations for the same input string).

Reply:
I added a little more detail here. Does this read more clearly, or is it too verbose?